src/module.common.ts

```diff
@@ -0,0 +1,141 @@
+import { NgModule, ModuleWithProviders } from '@angular/core';
```

Comment:
I'd rather see this be split up into `module.declarations.ts`, `module.imports.ts`, `module.providers.ts` and `module.bootstrap.ts`.

Its too many imports in one file IMO